Root.tsx

```diff
@@ -0,0 +1,84 @@
+import "react-native-gesture-handler";
+import { NavigationContainer } from "@react-navigation/native";
+import { createNativeStackNavigator } from "@react-navigation/native-stack";
+import HomeScreen from "./screens/home/HomeScreen";
+import RegisterScreen from "./screens/register/RegisterScreen";
+import type RootStackParamList from "./constants/navigation/RootStackParamListProps";
+import { useContext, useEffect, useState } from "react";
+import { AuthContext, AuthContextType } from "./context/AuthProvider";
+import { ActivityIndicator, View } from "react-native";
+import LoginScreen from "./screens/login/LoginScreen";
+import * as SecureStore from "expo-secure-store";
+import { setUserToken } from "./helpers/axios/axiosConfig";
+import styles from "./RootStyle";
+
+const Stack = createNativeStackNavigator<RootStackParamList>();
+
+const HomeStackNavigator = () => {
+  return (
+    <Stack.Navigator>
+      <Stack.Screen name="Home" component={HomeScreen} />
+    </Stack.Navigator>
+  );
+};
+
+const AuthStackNavigator = () => {
+  return (
+    <Stack.Navigator
+      screenOptions={{ headerShown: false, headerBackTitleVisible: false }}
+    >
+      <Stack.Screen
+        name="Login"
+        component={LoginScreen}
+        options={{ headerShown: false }}
+      />
+      <Stack.Screen
+        name="Register"
+        component={RegisterScreen}
+        options={{ headerShown: false }}
+      />
+    </Stack.Navigator>
+  );
+};
+
+export default function Root() {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const { user, setUser } = useContext<AuthContextType>(AuthContext);
+
+  useEffect(() => {
+    SecureStore.getItemAsync("user")
+      .then((userString) => {
+        if (userString) {
+          const user = JSON.parse(userString);
+          setUser(user);
+          setUserToken(user.token);
+        }
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  if (isLoading) {
+    return (
+      <View style={styles.activityIndicatorContainer}>
+        <ActivityIndicator size="large" color="#1d9bf1" />
+      </View>
+    );
+  }
+
+  return (
+    <>
+      {user ? (
+        <NavigationContainer>
+          <HomeStackNavigator />
+        </NavigationContainer>
+      ) : (
+        <NavigationContainer>
+          <AuthStackNavigator />
+        </NavigationContainer>
+      )}
+    </>
```

Comment:
it would be easier to follow to format this code as:

```
  <NavigationContainer>
      {user ? (
          <HomeStackNavigator />
      ) : (
          <AuthStackNavigator />
      )}
  </NavigationContainer>
```